### PR TITLE
[202205] Update SOC properties for DLR_INIT based pfcwd recovery

### PR DESCRIPTION
**Pull request by neethajohn**
#### Why I did it
Update soc properties for certain roles that need to use pfcwd dlr init based recovery mechanism

#### How to verify it
Updated the templates on a 7050cx3 dual tor and 7260 T1 which satisfies these conditions and validated pfcwd recovery which uses DLR_INIT based mechanism. Also validated that this mechanism is not used on 7050cx3 single tor with the updated templates

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

#### Tested branch (Please provide the tested image version)

- [x] 20220531.27 image with the updated bcm templates and custom swss deb

### CODE REVIEW

**Review by bingwang-ms — APPROVED** [review]

**Comment by bingwang-ms** [comment]
LGTM, except for the indention in the new block.
```
hybrid_pfc_deadlock_enable=1
                          pfc_deadlock_seq_control=1
                          sai_pfc_dlr_init_capability=1
```

**Comment by neethajohn** [comment]
> LGTM, except for the indention in the new block.
> 
> ```
> hybrid_pfc_deadlock_enable=1
>                           pfc_deadlock_seq_control=1
>                           sai_pfc_dlr_init_capability=1
> ```

The choices I had were to keep the readability in the j2 template vs final bcm config. I stuck to the first method. I also had an option to use a variable for each property vs a single variable. I thought it was better to use a single var since they all are needed together

File: src/sonic-config-engine/tests/sample_output/py3/arista7260-t1.config.bcm [hunk 1]
```diff
@@ -3,3 +3,6 @@ l3_alpm_hit_skip=1
 
 mmu_init_config="MSFT-TH2-Tier1"
 
+hybrid_pfc_deadlock_enable=1
+                          pfc_deadlock_seq_control=1
+                          sai_pfc_dlr_init_capability=1
```
Comment (lguohan) [inline src/sonic-config-engine/tests/sample_output/py3/arista7260-t1.config.bcm:8]: why do we have blanks here?

File: device/arista/x86_64-arista_7050cx3_32s/Arista-7050CX3-32S-C32/config.bcm.j2 [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
 {%- if DEVICE_METADATA is defined and DEVICE_METADATA['localhost'] is defined and DEVICE_METADATA['localhost']['subtype'] is defined -%}
 {%-     set switch_subtype = DEVICE_METADATA['localhost']['subtype'] -%}
 {%-     if 'dualtor' in switch_subtype.lower() %}
 {%-         set map_prio = 'sai_remap_prio_on_tnl_egress=1' -%}
 {%-     endif %}
 {%- endif %}
+{%- if SYSTEM_DEFAULTS is defined and 'tunnel_qos_remap' in SYSTEM_DEFAULTS and SYSTEM_DEFAULTS['tunnel_qos_remap']['status'] == 'enabled' -%}
+{%-     set pfcwd_sock = 'hybrid_pfc_deadlock_enable=1
+                          pfc_deadlock_seq_control=1
+                          sai_pfc_dlr_init_capability=1' -%}
```
Comment (lguohan) [inline device/arista/x86_64-arista_7050cx3_32s/Arista-7050CX3-32S-C32/config.bcm.j2:13]: `
{%-     set pfcwd_sock = 
'hybrid_pfc_deadlock_enable=1
pfc_deadlock_seq_control=1
sai_pfc_dlr_init_capability=1' -%}
`

Comment (neethajohn) [reply device/arista/x86_64-arista_7050cx3_32s/Arista-7050CX3-32S-C32/config.bcm.j2:13]: addressed in #15240

**Pull request merged**